### PR TITLE
initial value of ssthresh

### PR DESCRIPTION
**Pull request by scimbe**
Two points:

1) the initial value of  ssthresh = 65535 is usually too small. Therefore the slow start is limited too early,
 As long no limit (like send buffer size) is given the sky should be the limit.

rfc5681: The initial value of ssthresh SHOULD be set arbitrarily high (e.g., to the size of the largest possible advertised window)

2) Netperfmeter uses a kind of socketcall to manipulate the max send buffer (sendQueue size) by a dirty workaround ( tcpCommand->getUserId()).
However, after the send buffer is set, the state of the connection should be reinitialized (e.g. to set the initial value of the ssthresh)

### CODE REVIEW

File: src/transport/tcp/TCPConnection.h [hunk 1]
```diff
@@ -152,6 +152,7 @@ class INET_API TCPStateVariables : public cObject
     TCPStateVariables();
     virtual std::string info() const;
     virtual std::string detailedInfo() const;
+    virtual void reInitialState() {};
```
Comment (ZoltanBojthe) [inline src/transport/tcp/TCPConnection.h]: Would be better write a `virtual void setSendQueueLimit(uint32 newLimit) { sendQueueLimit = newLimit; }` function,
use it in `TCPConnection::process_QUEUE_BYTES_LIMIT()`,
and overwrite `setSendQueueLimit` in `TCPTahoeRenoFamilyStateVariables` class for update `sstresh`, too.

**Comment by scimbe** [comment]
Hope I hit it...

**Comment by rhornig** [comment]
This seems to be ok from our side. @tuexen could you give a final nod before merging?

**Comment by tuexen** [comment]
Setting the ssthresh to 0xffffffff makes sense.

**Comment by rhornig** [comment]
Squashed the commits and merged into the master. You can delete now your own topic branch now (or rebase on master)

**Comment by rhornig** [comment]
@scimbe Probably it would be good to include these changes also in the 'integration' branch. Changes in master will not be automatically merged into integration.

**Pull request closed without merging**